iterator.go

```diff
@@ -229,6 +235,12 @@ func (r RootLookuper) handleSuccessAnswer(ctx context.Context,
 		return "", resp, nil
 	}
 
+	if len(resp.Answer) == 0 && resp.Authoritative {
```

Comment:
we wouldn't have reached here if there were answers, maybe
```go
case exdns.HasNsType(resp, dns.TypeSOA) && resp.Authoritative:
    // reasons
    return "", resp, nil
```
directly in `doIteratePass()` instead of extending `handleSuccessAnswer`?